lib/neo4j/active_node/has_n.rb

```diff
@@ -29,11 +29,16 @@ def inspect
 
       extend Forwardable
       %w(include? empty? count find first last ==).each do |delegated_method|
-        def_delegator :@enumerable, delegated_method
+        def_delegator :deferred_objects_or_enumerable, delegated_method
       end
 
       include Enumerable
 
+      def deferred_objects_or_enumerable
+        # This is not good enough. Updates have to be taken care of as well
+        @query_proxy.start_object.try(:new_record?) ? @deferred_objects : @enumerable
```

Comment:
Regarding Chris' comment on `try`, is the `start_object` `nil` sometimes?  I can't imagine it would be something other than an `ActiveNode` object...